src/charm_state.py

```diff
@@ -114,7 +240,109 @@ def from_charm(cls, charm: CharmBase) -> "CharmConfig":
         else:
             openstack_clouds_yaml = None
 
-        return cls(runner_storage=runner_storage, openstack_clouds_yaml=openstack_clouds_yaml)
+        return cls(
+            path=path,
+            token=token,
+            reconcile_interval=reconcile_interval,
+            denylist=denylist,
+            dockerhub_mirror=dockerhub_mirror,
+            openstack_clouds_yaml=openstack_clouds_yaml,
+        )
+
+    @root_validator
+    @classmethod
+    def check_fields(cls, values: dict) -> dict:
+        """Validate the general charm configuration."""
+        reconcile_interval = values.get("reconcile_interval")
+        # By property definition, this cannot be None. Needed for mypy type check.
+        assert reconcile_interval is not None, "Unreachable code"  # nosec for [B101:assert_used]
```

Comment:
```suggestion
        reconcile_interval = values["reconcile_interval"]
```
If it cannot be None, this should work and mypy should not complain (at least I was able to run `tox -e lint` without errors).